assets/js/modules/subscribe-with-google/components/common/ProductsInput.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Subscribe with Google Account Create component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { TextField, Input } from '../../../../material-components';
+import { isValidProducts } from '../../util/validation';
+const { useDispatch, useSelect } = Data;
+
+export default function ProductsInput() {
+	// Get value.
+	const products = useSelect( ( select ) =>
+		select( STORE_NAME ).getProducts()
+	);
+	const productsString = products.join( '\n' );
+
+	// Handle form input.
+	const { setProducts } = useDispatch( STORE_NAME );
+	const onChange = useCallback(
+		( { currentTarget } ) => {
+			setProducts( currentTarget.value.split( '\n' ) );
+		},
+		[ setProducts ]
+	);
+
+	return (
+		<TextField
+			className={ classnames( {
+				'mdc-text-field--error': ! isValidProducts( products ),
```

Comment:
The visual error indicator currently shows even when you haven't entered anything (i.e. on page load), which is a slightly confusing experience.

![Screen Shot 2021-08-23 at 11 26 52 AM](https://user-images.githubusercontent.com/3531426/130497972-cb9a1816-0f25-4bdb-a11f-e56bd1521609.png)

We typically handle this by only showing a visual error indicator when what the user has entered is non-empty and invalid. Basically, the `isValidProducts` function should remain as is (for example the user should still not be able to proceed when not entering anything), but we need to add an extra `! products || ! products.length || ...` check here so that no error shows on page load (or generally when nothing is entered.